src/components/user/repo/db/trashbin.schema.js

```diff
@@ -0,0 +1,23 @@
+const mongoose = require('mongoose');
+
+const { Schema } = mongoose;
+
+const schema = new Schema(
```

Comment:
is there a generic versioning available instead which can be used for restoration? how restrict and allow access with that model? how is that planned?

Reply:
generic versioning is a completely different usecase, because it is for when the data still actually is valid. On deletion, we have to remove all prior versions too, in theory including our backups.

How to allow access: not at all for now. How to restrict: dont implement read operations, only user support can access it manually for now.
Later obviously the user or admin will be able to restore the data.